src/main/java/com/bankingservice/banking/services/AccountService.java

```diff
@@ -35,11 +38,16 @@ public class AccountService {
      * @param registerRequestDTO
      * @return registerUserResponseDTO
      */
-    public RegisterUserResponseDTO insertDetailsForRegistration(RegisterRequestDTO registerRequestDTO) {
-        logger.info("[insertDetailsForRegistration] Registering a new user: {}", registerRequestDTO);
-        RegisterUserModel entity = registerUserRepository.save(accountServiceHelper.makeRegisterEntity(registerRequestDTO));
-        RegisterUserResponseDTO registerUserResponseDTO = accountServiceHelper.makeRegisterUserResponseDTO(entity);
-        return registerUserResponseDTO;
+    public RegisterUserResponseDTO insertDetailsForRegistration(RegisterRequestDTO registerRequestDTO) throws InsertionFailedException {
+        try {
+            logger.info("[insertDetailsForRegistration] Registering a new user: {}", registerRequestDTO);
+            RegisterUserModel entity = registerUserRepository.save(accountServiceHelper.makeRegisterEntity(registerRequestDTO));
+            RegisterUserResponseDTO registerUserResponseDTO = accountServiceHelper.makeRegisterUserResponseDTO(entity);
+            return registerUserResponseDTO;
+        } catch (Exception e) {
```

Comment:
1. Use the specific exception which can be thrown by specific method calls.
2. when catching an exception dont use debug logs, instead use error logs, debugs logs are made for the data which might be reuired while debuging in future, so that you might not require to add at that time